src/components/v2/devtronStackManager/DevtronStackManager.tsx

```diff
@@ -0,0 +1,441 @@
+     * when on module details view
+     */
+    useEffect(() => {
+        if (stackDetails.discoverModulesList.length > 0) {
+            if (
+                location.pathname.includes(URLS.STACK_MANAGER_INSTALLED_MODULES_DETAILS) &&
+                stackDetails.installedModulesList.length === 0
+            ) {
+                history.push(URLS.STACK_MANAGER_INSTALLED_MODULES)
+            } else if (location.pathname.includes('/details') && queryParams.get('id')) {
+                const _selectedModule = stackDetails.discoverModulesList.find(
+                    (module) => module.name.toLowerCase() === queryParams.get('id').toLowerCase(),
+                )
+
+                if (!_selectedModule) {
+                    history.push(
+                        location.pathname.includes(URLS.STACK_MANAGER_DISCOVER_MODULES_DETAILS)
+                            ? URLS.STACK_MANAGER_DISCOVER_MODULES
+                            : URLS.STACK_MANAGER_INSTALLED_MODULES,
+                    )
+                }
+
+                setSelectedModule(_selectedModule)
+                if (!detailsMode) {
+                    setDetailsMode(
+                        location.pathname.includes(URLS.STACK_MANAGER_DISCOVER_MODULES_DETAILS)
+                            ? 'discover'
+                            : 'installed',
+                    )
+                }
+            }
+        }
+    }, [stackDetails.discoverModulesList])
+
+    /**
+     * 1. If query params has 'id' then module installation action has been triggered
+     * so fetch the specific module info.
+     * 2. Else it's the server/stack upgrade action so fetch the server info
+     */
+    const getLatestInfo = async () => {
+        try {
+            if (queryParams.has('id')) {
+                const { result } = await getModuleInfo(queryParams.get('id'))
+
+                if (result) {
+                    const currentModule = stackDetails.discoverModulesList.find(
+                        (_module) => _module.name === result.name,
+                    )
+                    setSelectedModule({
+                        ...currentModule,
+                        installationStatus: result.status,
+                    })
+                }
+            } else {
+                getCurrentServerInfo()
+            }
+        } catch (e) {
+            showError(e)
+        } finally {
+            setActionTriggered({
+                ...actionTriggered,
+                [location.pathname.includes(URLS.STACK_MANAGER_DISCOVER_MODULES_DETAILS)
+                    ? `moduleAction-${queryParams.get('id')?.toLowerCase()}`
+                    : 'serverAction']: false,
+            })
+        }
+    }
+
+    const _getLogPodName = async (): Promise<void> => {
+        try {
+            const { result } = await getLogPodName()
+            setLogPodName(result?.podName)
+        } catch (e) {
+            console.error(e)
```

Comment:
we should remove console.error

Reply:
@vivek-devtron As discussed, provided specific error messages instead of console logging the whole error.